src/drivers/net/bnxt/bnxt.c

```diff
@@ -2172,6 +2172,7 @@ static int bnxt_tx ( struct net_device *dev, struct io_buffer *iob )
 
 	if ( bnxt_tx_avail ( bp ) < 1 ) {
 		DBGP ( "- %s (  ): Failed no bd's available\n", __func__ );
+		udelay ( 10000 );
```

Comment:
Why is a delay necessary here?  When there are no available transmit descriptors, the routine should either drop the packet, or use `netdev_tx_defer()` to enqueue it for automatic retransmission as soon as the next transmit completion is reported by the driver.

The only devices that need to use `netdev_tx_defer()` are those with very small transmit rings (e.g. RTL8139, where the legacy mode of operation allows for only 4 transmit descriptors).  Most drivers just drop packets, which is typically the correct thing to do: the only time we end up with high transmit volumes is when sending TCP ACKs, and it's generally better to harmlessly drop ACKs than to build up a large transmit queue.

Reply:
Agreed, I have removed these unnecessary delays.